### PR TITLE
Add defaultLassoAssertionTimeout

### PR DESCRIPTION
**Pull request by g-mark**
## Describe your changes

Adds a way to configure the time out for view life cycle assertions (`assertPresentation`, `assertRoot`, etc.).

This configuration is _per XCTestCase subclass_.  For more granular control, the pre-existing `timeout` parameters can be used.

Add `AssertionTimeoutOverride` conformance to your test case, and provide a value for `defaultLassoAssertionTimeout` to use a custom timeout.

### CODE REVIEW

**Review by russellsavageww — APPROVED** [review]
Looks good to me

File: Sources/LassoTestUtilities/ControllerLifecycle.swift [hunk 1]
```diff
@@ -19,8 +19,32 @@ import XCTest
...
     
+    /// The effective timeout to use when waiting on the main queue.
+    ///
+    /// By default all timeouts are one second.
+    ///
+    /// Add `AssertionTimeoutOverride` conformance to your test case, and provide
+    /// a value for `defaultLassoAssertionTimeout` to use a custom timeout.
+    internal var lassoAssertionTimeout: TimeInterval {
+        if let override = self as? AssertionTimeoutOverride {
+            return override.defaultLassoAssertionTimeout
```
Comment (DangWW) [inline Sources/LassoTestUtilities/ControllerLifecycle.swift:43]: less of a lasso comment and more of a, "wow, i did not know you could use keywords like that other than `self`"

Comment (g-mark) [reply Sources/LassoTestUtilities/ControllerLifecycle.swift:43]: Lol. Yep - in that context it's just a var name!  Git's syntax highlighting is too simplistic to reflect that :/

**Review by Ian-L — APPROVED** [review]
I couldn't spot any issues. Thanks @g-mark

**Pull request merged**